### ISSUE TITLE
With the `generateWebDocumentation` script, there is a failure during files copy

### ISSUE THREAD

**Issue by pylapp** [Orange-OpenSource/ouds-ios#1303]
# Description

The script generating the HTML documentation and uploading it does not not work anymore.
Maybe there are evolutions in the *swift* command dedicated to the documentation.

# Logs

Logs after the generation of the documentation with the *swift* command:

```text
Generated combined documentation archive at:
  /tmp/ouds-docs-1770973523
++ find /tmp/ouds-docs-1770973523 -type f
++ wc -l
++ xargs
+ files_count=53969
+ _ '👍 Generated '\''53969'\'' files!'
+ local 'message=👍 Generated '\''53969'\'' files!'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 👍 Generated '\''53969'\'' files!'
🍊 👍 Generated '53969' files!
+ [[ 1 -eq 1 ]]
+ _ '👉 Updating CNAME file'
+ local 'message=👉 Updating CNAME file'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 👉 Updating CNAME file'
🍊 👉 Updating CNAME file
+ echo ios.unified-design-system.orange.com
+ _ '👍 Updated!'
+ local 'message=👍 Updated!'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 👍 Updated!'
🍊 👍 Updated!
+ echo '<!doctype html><html><head><meta http-equiv="refresh" content="0; URL= https://ios.unified-design-system.orange.com/documentation/oudsthemescontract"></head><body>Redirecting to https://ios.unified-design-system.orange.com/documentation/oudsthemescontract</body></html>'
+ [[ 1 -eq 1 ]]
+ _ '👉 Versioning documentation in service pages branch (it can take a lot of time)...'
+ local 'message=👉 Versioning documentation in service pages branch (it can take a lot of time)...'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 👉 Versioning documentation in service pages branch (it can take a lot of time)...'
🍊 👉 Versioning documentation in service pages branch (it can take a lot of time)...
+ clean_repo
+ _ '🧹 Cleaning Git repository'
+ local 'message=🧹 Cleaning Git repository'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 🧹 Cleaning Git repository'
🍊 🧹 Cleaning Git repository
+ git reset --hard
HEAD est maintenant à 82b59181 ci: improve management of directory to remove to generate doc
+ git clean -fd
+ git prune
+ _ '🔨 Checkout service pages branch, align with remote'
+ local 'message=🔨 Checkout service pages branch, align with remote'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 🔨 Checkout service pages branch, align with remote'
🍊 🔨 Checkout service pages branch, align with remote
+ git show-ref --verify --quiet refs/heads/gh-pages
+ _ '🔨 Local branch '\''gh-pages'\'' does not exist. Checking out from remote.'
+ local 'message=🔨 Local branch '\''gh-pages'\'' does not exist. Checking out from remote.'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 🔨 Local branch '\''gh-pages'\'' does not exist. Checking out from remote.'
🍊 🔨 Local branch 'gh-pages' does not exist. Checking out from remote.
+ git fetch origin
+ git checkout -b gh-pages origin/gh-pages
Mise à jour des fichiers: 100% (51764/51764), fait.
la branche 'gh-pages' est paramétrée pour suivre 'origin/gh-pages'.
Basculement sur la nouvelle branche 'gh-pages'
+ _ '🔨 Applying changes'
+ local 'message=🔨 Applying changes'
+ local is_error=false
+ local 'prefix=🍊 '
+ [[ false == \t\r\u\e ]]
+ echo '🍊 🔨 Applying changes'
🍊 🔨 Applying changes
+ clean_directory css
+ '[' -d 99581 ']'
+ clean_directory data
+ '[' -d 99581 ']'
+ clean_directory documentation
+ '[' -d 99581 ']'
+ clean_directory images
+ '[' -d 99581 ']'
+ clean_directory img
+ '[' -d 99581 ']'
+ clean_directory index
+ '[' -d 99581 ']'
+ clean_directory js
+ '[' -d 99581 ']'
+ clean_directory '*.jpg'
+ '[' -d 99581 ']'
+ clean_directory .html
+ '[' -d 99581 ']'
+ clean_directory CNAME
+ '[' -d 99581 ']'
+ cp -r /tmp/ouds-docs-1770973523/CNAME /tmp/ouds-docs-1770973523/css /tmp/ouds-docs-1770973523/data /tmp/ouds-docs-1770973523/developer-og-twitter.jpg /tmp/ouds-docs-1770973523/developer-og.jpg /tmp/ouds-docs-1770973523/documentation /tmp/ouds-docs-1770973523/downloads /tmp/ouds-docs-1770973523/favicon.ico /tmp/ouds-docs-1770973523/favicon.svg /tmp/ouds-docs-1770973523/images /tmp/ouds-docs-1770973523/img /tmp/ouds-docs-1770973523/index /tmp/ouds-docs-1770973523/index.html /tmp/ouds-docs-1770973523/js /tmp/ouds-docs-1770973523/link-hierarchy.json /tmp/ouds-docs-1770973523/linkable-entities.json /tmp/ouds-docs-1770973523/metadata.json /tmp/ouds-docs-1770973523/tutorials /tmp/ouds-docs-1770973523/videos ..
./generateWebDocumentation.sh: line 326: 12213 Killed: 9               cp -r "$DOCUMENTATION_HTML_LOCATION"/* "$DOCS_DIRECTORY"
++ on_error_signal
++ _ '❌  An error occurred with command '\''cp -r "$DOCUMENTATION_HTML_LOCATION"/* "$DOCS_DIRECTORY"'\''. Exits. (1)'
++ local 'message=❌  An error occurred with command '\''cp -r "$DOCUMENTATION_HTML_LOCATION"/* "$DOCS_DIRECTORY"'\''. Exits. (1)'
++ local is_error=false
++ local 'prefix=🍊 '
++ [[ false == \t\r\u\e ]]
++ echo '🍊 ❌  An error occurred with command '\''cp -r "$DOCUMENTATION_HTML_LOCATION"/* "$DOCS_DIRECTORY"'\''. Exits. (1)'
🍊 ❌  An error occurred with command 'cp -r "$DOCUMENTATION_HTML_LOCATION"/* "$DOCS_DIRECTORY"'. Exits. (1)
++ [[ 1 -eq 1 ]]
++ clean_repo
++ _ '🧹 Cleaning Git repository'
++ local 'message=🧹 Cleaning Git repository'
++ local is_error=false
++ local 'prefix=🍊 '
++ [[ false == \t\r\u\e ]]
++ echo '🍊 🧹 Cleaning Git repository'
🍊 🧹 Cleaning Git repository
++ git reset --hard
./generateWebDocumentation.sh: line 85: 12218 Killed: 9               [[ "$is_error" == "true" ]]
```

# Workaround

- Get the generate files form the output temporary folder and copy them to the *gh-pages* branch 

# Steps

- [ ] Add the *x* option in the `set -euo pipefail`
- [ ] Run the command, check logs
- [ ] Check the errors and why it occurs
- [ ] Fix it
- [ ] Retry the command, no errors must appear

**Comment by B3nz01d** [comment]
Issue finished during Sprint # 28
